### PR TITLE
Newsletters: add 289 (2024-02-14)

### PR DESCRIPTION
**Pull request by harding**
- [ ] Lede, releases/RCs, topic entries @harding 
- [x] Bitcoin Core PR Review Club @LarryRuane

### CODE REVIEW

File: _posts/en/newsletters/2024-02-14-newsletter.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+  the master private key.  When referring to just a path, the `m` is
+  unnecessary and may be wrong in some contexts.
+
+{% assign day_after_posting = page.date | date: "%s" | plus: 86400 | date: "%Y-%m-%d 15:00" %}
+{% include snippets/recap-ad.md when=day_after_posting %}
+{% include references.md %}
+{% include linkers/issues.md v=2 issues="28948,6785,2818,2816,8338,2856,2442,2451,1131,1096" %}
+[fsfe]: https://fsfe.org/activities/ilovefs/index.en.html
+[fsf]: https://www.fsf.org/blogs/community/i-love-free-software-day-is-here-share-your-love-software-and-a-video
+[sanders future]: https://delvingbitcoin.org/t/v3-and-some-possible-futures/523/2
```
Comment (bitschmidty) [inline _posts/en/newsletters/2024-02-14-newsletter.md:196]:
```suggestion
[sanders future]: https://delvingbitcoin.org/t/v3-and-some-possible-futures/523
```

**Comment by kanzure** [comment]
https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2024-February/022327.html

File: _posts/en/newsletters/2024-02-14-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+  within the next block or two.
+
+- **What would have happened if v3 semantics had been applied to anchor outputs a year ago?**
+  Suhas Daftuar [posted][daftuar retrospective] to Delving Bitcoin about
+  his research into the idea of automatically applying [v3 transaction
+  relay policy][topic v3 transaction relay] to [anchors-style][topic
+  anchor outputs] LN commitment and fee-bumping transactions (see
+  [Newsletter #286][news286 imbued] for the underlying _imbued v3_
+  proposal).  In short, he recorded 14,124 transactions in 2023 that
+  looked like anchor spends.  Of those,
```
Comment (bitschmidty) [inline _posts/en/newsletters/2024-02-14-newsletter.md:32]: Maybe

```suggestion
  looked like anchor spends.  Of those:
```

File: _posts/en/newsletters/2024-02-14-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+  - About 94% <!-- (14124 - 856) / 14124 --> would have been successful
+    under the v3 rules.
+
+  - About 2.1% <!-- 302/14124 --> had more than one parent (e.g.,
+    attempts to batch CPFP spends).  LN wallets do this for efficiency
```
Comment (bitschmidty) [inline _posts/en/newsletters/2024-02-14-newsletter.md:38]:
```suggestion
    attempts to batch [CPFP][topic cpfp] spends).  LN wallets do this for efficiency
```

File: _posts/en/newsletters/2024-02-14-newsletter.md [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    resulting in the anchor spending having more than the allowed one
+    parent.  Developer Bastien Teinturier thinks that this may have been
+    a behavior of Eclair and notes that Eclair would resolve this
+    situation automatically even with its current code.
+
+  - Less than 0.1% <!-- 10/14124 --> were larger than 1,000 vbytes.
+    This is also behavior that LN wallets would need to change.
+    Daftuar's further research showed that nearly all anchor spends were
+    less than 500 vbytes, potentially suggesting that the v3 size limit
+    could be reduced.  This would make it less expensive for a defender
```
Comment (bitschmidty) [inline _posts/en/newsletters/2024-02-14-newsletter.md:68]: Relevant?
https://bitcoinops.org/en/newsletters/2024/01/03/#v3-transaction-pinning-costs

File: _posts/en/newsletters/2024-02-14-newsletter.md [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
+[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
+Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
+Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
+Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo],
+[Bitcoin Inquisition][bitcoin inquisition repo], and [BINANAs][binana
+repo]._
+
+- [Bitcoin Core #28948][] adds support for [version 3 transaction
+  relay][topic v3 transaction relay] allowing any v3 transaction that
```
Comment (bitschmidty) [inline _posts/en/newsletters/2024-02-14-newsletter.md:127]:
```suggestion
  relay][topic v3 transaction relay], allowing any v3 transaction that
```

File: _posts/en/newsletters/2024-02-14-newsletter.md [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+  automatically create other unrelated transactions using that input.
+  However, if that transaction becomes unconfirmable because a different
+  input in that transaction was double spent, Bitcoin Core's wallet will
+  automatically allow the UTXO to be spent in a different transaction
+  again.  Unfortunately, if a parent of the transaction is made
+  unconfirmable because a different version was confirmed, Bitcoin
+  Core's wallet will not currently automatically allow the UTXO to be
+  spent.  Eclair can independently detect a double spend of the parent
+  transaction and it will now tell Bitcoin Core's wallet to manually
+  unlock the UTXO so it can be spent again.
```
Comment (bitschmidty) [inline _posts/en/newsletters/2024-02-14-newsletter.md:160]:
```suggestion
  abandon the transaction so the UTXO can be spent again.
```

And perhaps link to https://developer.bitcoin.org/reference/rpc/abandontransaction.html

File: _posts/en/newsletters/2024-02-14-newsletter.md [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+layout: newsletter
+lang: en
+---
+This week's newsletter FIXME:harding
+
+## News
+
+- **Ideas for relay enhancements after cluster mempool is deployed:**
+  Gregory Sanders [posted][sanders future] to Delving Bitcoin several
+  ideas for allowing individual transactions to opt-in certain mempool
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-14-newsletter.md:16]:
```suggestion
  ideas for allowing individual transactions to opt-in to certain mempool
```

File: _posts/en/newsletters/2024-02-14-newsletter.md [hunk 8]
```diff
@@ -0,0 +1,283 @@
...
+  [Newsletter #286][news286 imbued] for the underlying _imbued v3_
+  proposal).  In short, he recorded 14,124 transactions in 2023 that
+  looked like anchor spends.  Of those,
+
+  - About 94% <!-- (14124 - 856) / 14124 --> would have been successful
+    under the v3 rules.
+
+  - About 2.1% <!-- 302/14124 --> had more than one parent (e.g.,
+    attempts to batch CPFP spends).  LN wallets do this for efficiency
+    when closing more than channel within a short amount of time.  They
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-14-newsletter.md:39]:
```suggestion
    when closing more than one channel within a short amount of time.  They
```

File: _posts/en/newsletters/2024-02-14-newsletter.md [hunk 9]
```diff
@@ -0,0 +1,283 @@
...
+    to overcome an attempted [pinning attack][topic transaction pinning]
+    against an anchor spend, but it would also prevent LN wallets from
+    being able to contribute fees from more than a few UTXOs.
+    Teinturier [noted][teinturier better] that "it's very tempting to
+    reduce the 1,000 vbytes value, but past data only shows honest
+    attempts (with very few pending HTLCs) as we haven’t seen any
+    widespread attacks on the network yet, so it’s hard to figure out
+    what value would be 'better'."
+
+  Although additional discussion and research on this topic is expected,
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-14-newsletter.md:78]:
```suggestion
  Although additional discussion and research on this topic are expected,
```

File: _posts/en/newsletters/2024-02-14-newsletter.md [hunk 10]
```diff
@@ -0,0 +1,283 @@
...
+
+- [Core Lightning #6785][] makes [anchor-style][topic anchor outputs]
+  channels the default on Bitcoin.  Non-anchor channels are still used
+  for channels on Elements-compatible [sidechains][topic sidechains].
+
+- [Eclair #2818][] maximizes the number of inputs the Eclair wallet
+  believes it can safely spend by detecting some cases when an existing
+  unconfirmed transaction is very unlikely to become confirmed.  Eclair
+  uses Bitcoin Core's wallet to manage its UTXOs for onchain spending,
+  including for fee bumping transactions.  When a UTXO controlled by the
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-14-newsletter.md:228]:
```suggestion
  including for fee-bumping transactions.  When a UTXO controlled by the
```

File: _posts/en/newsletters/2024-02-14-newsletter.md [hunk 11]
```diff
@@ -0,0 +1,283 @@
...
+  again.  Unfortunately, if a parent of the transaction is made
+  unconfirmable because a different version was confirmed, Bitcoin
+  Core's wallet will not currently automatically allow the UTXO to be
+  spent.  Eclair can independently detect a double spend of the parent
+  transaction and it will now tell Bitcoin Core's wallet to manually
+  unlock the UTXO so it can be spent again.
+
+- [Eclair #2816][] allows the node operator to choose the maximum amount
+  they're willing to spend on an [anchor output][topic anchor outputs]
+  in order to get a commitment transaction confirmed.  Previously Eclair
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-14-newsletter.md:243]:
```suggestion
  to get a commitment transaction confirmed.  Previously Eclair
```

File: _posts/en/newsletters/2024-02-14-newsletter.md [hunk 12]
```diff
@@ -0,0 +1,283 @@
...
+  would spend up to 5% of the channel value, but that may be too high
+  for high-value channels.  Eclair's new default is the maximum feerate
+  suggested by its feerate estimator, up to an absolute total of 10,000
+  sat.  Eclair will also still pay up to the amount at risk from
+  [HTLCs][topic htlc] expiring in the near future, which could be higher
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-14-newsletter.md:248]:
```suggestion
  [HTLCs][topic htlc] expiring soon, which could be higher
```

File: _posts/en/newsletters/2024-02-14-newsletter.md [hunk 13]
```diff
@@ -0,0 +1,283 @@
...
+Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
+Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo],
+[Bitcoin Inquisition][bitcoin inquisition repo], and [BINANAs][binana
+repo]._
+
+- [Bitcoin Core #28948][] adds support for [version 3 transaction
+  relay][topic v3 transaction relay] allowing any v3 transaction that
+  has no unconfirmed parent to enter the mempool according to the normal
+  transaction acceptance rules.  The v3 transaction can be [CPFP fee
+  bumped ][topic cpfp] but only if the child is 1,000 vbytes or less.
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-14-newsletter.md:209]:
```suggestion
  transaction acceptance rules.  The v3 transaction can be [CPFP
  fee-bumped][topic cpfp] but only if the child is 1,000 vbytes or less.
```

File: _posts/en/newsletters/2024-02-14-newsletter.md [hunk 14]
```diff
@@ -0,0 +1,283 @@
...
+  Each v3 parent may only have one unconfirmed child transaction in the
+  mempool and each child may only have one unconfirmed parent.  Either
+  the parent or child transaction can always be [replaced by fee][topic
+  rbf].  The rules only apply to Bitcoin Core's relay policy; at the
+  consensus layer, v3 transactions are validated the same as version 2
+  transactions defined in [BIP68][].  The new rules are intended to help
+  contract protocols such as LN ensure their precommitted transactions
+  can always be confirmed quickly with minimal extra fees needed to
+  escape [transaction pinning][topic transaction pinning].
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-14-newsletter.md:218]:
```suggestion
  escape [transaction pinning][topic transaction pinning] attacks.
```

File: _posts/en/newsletters/2024-02-14-newsletter.md (hunk 4 above)
Comment (harding) [reply _posts/en/newsletters/2024-02-14-newsletter.md:68]: Yes, but IIRC a couple reviewers on that newsletter found my description a bit hard to follow, so I think I'm going to stick with just the link to the pinning topic page.  Thanks for suggesting and finding that link, though!

File: _posts/en/newsletters/2024-02-14-newsletter.md (hunk 14 above)
Comment (harding) [reply _posts/en/newsletters/2024-02-14-newsletter.md:218]: Leaving as-is.  There are legit cases where transaction pinning can happen by accident, especially if CPFP carve out wasn't a thing (which it might not be sooner or later), so pinning isn't always an attack.

File: _posts/en/newsletters/2024-02-14-newsletter.md [hunk 15]
```diff
@@ -0,0 +1,295 @@
...
+Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
+[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
+Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
+Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
+Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo],
+[Bitcoin Inquisition][bitcoin inquisition repo], and [BINANAs][binana
+repo]._
+
+- [Bitcoin Core #28948][] adds support for (but does not enable) [version 3 transaction
+  relay][topic v3 transaction relay], allowing any v3 transaction that
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-14-newsletter.md:213]: Is it worth mentioning that the child (of a v3 transaction) does _not_ have to also be v3? Is that true? If the child _does_ have to also be v3, that might be worth mentioning.

Wait, what is the rule exactly? I couldn't tell from the PR description. Maybe it is:

> An unconfirmed v3 tx can have _either_ one unconfirmed parent _or_ one unconfirmed child (not both), and that parent or child can also be v3 but doesn't have to be. An unconfirmed v3 tx can also have no unconfirmed parents or children.

If that's correct, maybe consider stating that; it would have helped me understand it more easily.

**Comment by harding** [comment]
Added lede, removed releases/RCs section (none this week), and added topic entries.  Reviewed review club section (thanks @LarryRuane!) and made all edits (thanks @glozow @glozow @LarryRuane !).  Thanks everyone!

**Review by bitschmidty — APPROVED** [review]
ACK topics, lede, and PR Review section updates

**Comment by bitschmidty** [comment]
Squashed and merged, thank you @harding, @LarryRuane and team! 🚀

File: _posts/en/newsletters/2024-02-14-newsletter.md [hunk 16]
```diff
@@ -0,0 +1,295 @@
...
+      high-fee rate C is added which spends from both A and B.
+      (This is an allowed package topology because it's only two levels,
+      although it was pointed out that the `submitpackage` RPC doesn't allow
+      this topology.) In this case, C would be accepted because much of its
+      fee is absorbed by A and B, but C should be rejected."
+  a5link="https://bitcoincore.reviews/28950#l-108"
+
+  q6="Why can’t `maxfeerate` be checked immediately after decoding like
+      `maxburnamount` is?"
+  a6="Because transaction inputs famously don't explicitly state the
```
Comment (glozow) [inline _posts/en/newsletters/2024-02-14-newsletter.md:181]: famously 🤣

File: _posts/en/newsletters/2024-02-14-newsletter.md (hunk 15 above)
Comment (glozow) [reply _posts/en/newsletters/2024-02-14-newsletter.md:213]: the child of a v3 tx *does* need to be v3.

File: _posts/en/newsletters/2024-02-14-newsletter.md (hunk 16 above)
Comment (LarryRuane) [reply _posts/en/newsletters/2024-02-14-newsletter.md:181]: Maybe I shouldn't have tried to be cute there, it just kind of slipped out, wasn't my fault! :grin:

**Comment by ariard** [comment]
appreciated if there is a correction in future newsletters on Bitcoin Core #28948.
the merge has not been consensual according to Core review standards and the anti-pinning mitigation is weak
cf. my comment here: https://github.com/bitcoin/bitcoin/pull/28948#issuecomment-1946770281
